main.cpp

```diff
@@ -104,6 +114,394 @@ good to go!
 
 #include <iostream>
 
+struct FloatType;
+struct DoubleType;
+struct IntType;
+
+struct FloatType
+{
+    float* value = nullptr;
+
+    FloatType(float _value)
+    {
+        value = new float();
+        *value = _value;
+    }
+    ~FloatType()
+    {
+        if ( value != nullptr )
+            delete value;
+    }
+
+    FloatType& add( float operand )
+    {
+        if (value != nullptr)    
+            *value = *value + operand;
+
+        return *this;
+    }
+
+    FloatType& add(const FloatType& ft);
+    FloatType& add(const DoubleType& ft);
+    FloatType& add(const IntType& ft);
+
+    FloatType& subtract( float operand )
+    {
+        if (value != nullptr)   
+            *value = *value - operand;
+
+        return *this;
+    }
+
+    FloatType& subtract(const FloatType& ft);
+    FloatType& subtract(const DoubleType& ft);
+    FloatType& subtract(const IntType& ft);
+
+    FloatType& multiply( float operand )
+    {
+        if (value != nullptr) 
+            *value = *value * operand;
+
+        return *this;
+    }
+
+    FloatType& multiply(const FloatType& ft);
+    FloatType& multiply(const DoubleType& ft);
+    FloatType& multiply(const IntType& ft);
+
+    FloatType& divide( float operand )
+    {
+        if (!(operand > 0 || operand < 0))
+            std::cout << "warning: floating point division by zero!" << std::endl; 
```

Comment:
You can check if the `operand != 0.f`
no need for this complex comparison.
- [ ] revise